test/unit/Ramm/swap.js

```diff
@@ -404,4 +405,126 @@ describe('swap', function () {
+    const { ramm, pool, mcr, tokenController } = fixture.contracts;
+    const [member] = fixture.accounts.members;
+
+    const ethIn = parseEther('1');
+    const minAmountOut = parseEther('28.8');
+
+    const { timestamp } = await ethers.provider.getBlock('latest');
+    const nextBlockTimestamp = timestamp + 5 * 60;
+    const deadline = nextBlockTimestamp + 5 * 60;
+
+    const state = await getStateAtBlockTimestamp(ramm, pool, mcr, tokenController, nextBlockTimestamp);
+    const { nxmOut } = getExpectedStateAfterSwapEthForNxm(state, ethIn);
+
+    await setNextBlockTime(nextBlockTimestamp);
+    const swap = ramm.connect(member).swap(0, minAmountOut, deadline, { value: ethIn, maxPriorityFeePerGas: 0 });
+    await expect(swap).to.emit(ramm, 'EthSwappedForNxm').withArgs(member.address, ethIn, nxmOut);
+  });
+
+  it('should revert when the eth circuit breaker is hit', async function () {
+    const fixture = await loadFixture(setup);
+    const { ramm, pool, mcr, tokenController } = fixture.contracts;
+    const [member] = fixture.accounts.members;
+    const { emergencyAdmin } = fixture.accounts;
+
+    const nxmIn = parseEther('1000');
+    const minAmountOut = parseEther('0');
+
+    const { timestamp } = await ethers.provider.getBlock('latest');
+    const nextBlockTimestamp = timestamp + 10;
+    const deadline = nextBlockTimestamp + 5 * 60;
+
+    const state = await getStateAtBlockTimestamp(ramm, pool, mcr, tokenController, nextBlockTimestamp);
+    const { ethOut } = getExpectedStateAfterSwapNxmForEth(state, nxmIn);
+
+    await ramm.connect(emergencyAdmin).setCircuitBreakerLimits(ethOut.div(WeiPerEther).sub(1), 0);
+
+    await setNextBlockTime(nextBlockTimestamp);
+    await expect(ramm.connect(member).swap(nxmIn, minAmountOut, deadline)).to.revertedWithCustomError(
+      ramm,
+      'EthCircuitBreakerHit',
+    );
+  });
+
+  it('should revert when nxm circuit breaker is hit', async function () {
+    const fixture = await loadFixture(setup);
+    const { ramm, pool, mcr, tokenController } = fixture.contracts;
+    const [member] = fixture.accounts.members;
+    const { emergencyAdmin } = fixture.accounts;
+
+    const ethIn = parseEther('1');
+    const minAmountOut = parseEther('0');
+
+    const { timestamp } = await ethers.provider.getBlock('latest');
+    const nextBlockTimestamp = timestamp + 10;
+    const deadline = nextBlockTimestamp + 5 * 60;
+
+    const state = await getStateAtBlockTimestamp(ramm, pool, mcr, tokenController, nextBlockTimestamp);
+    const { nxmOut } = getExpectedStateAfterSwapEthForNxm(state, ethIn);
+
+    await ramm.connect(emergencyAdmin).setCircuitBreakerLimits(nxmOut.div(WeiPerEther).sub(1), 0);
+
+    await setNextBlockTime(nextBlockTimestamp);
+    const swap = ramm.connect(member).swap(0, minAmountOut, deadline, { value: ethIn, maxPriorityFeePerGas: 0 });
+    await expect(swap).to.emit(ramm, 'EthSwappedForNxm').withArgs(member.address, ethIn, nxmOut);
```

Comment:
should be:
```javascript
    ....to.revertedWithCustomError(ramm, 'NxmCircuitBreakerHit');
```

Also I was expecting the swapped events to not be emitted if it was reverted. We should probably move the emit swapped events after the else clause of the revert circuit breakers

Reply:
Events are not emitted if the tx reverts so their location doesn't really matter.

Reply:
Fixed the test nevertheless, it was not checking what it had to check. It wasn't reverting because I didn't set the limit for the nxm (it's the second parameter, not the first)